MerlinAU.sh

```diff
@@ -3635,24 +3664,50 @@ _ChangelogVerificationCheck_()
     fi
 }
 
-##------------------------------------------##
-## Modified by ExtremeFiretop [2024-May-25] ##
-##------------------------------------------##
+##----------------------------------------##
+## Modified by Martinski W. [2024-May-31] ##
+##----------------------------------------##
 _ManageChangelog_()
 {
-    local mode="$1"  # Mode should be 'download' or 'view'
-    local wgetLogFile changeLogTag changeLogFile changeLogURL
+    if [ $# -eq 0 ] || [ -z "$1" ]
+    then echo "**ERROR** **NO_PARAMS**" ; return 1 ; fi
+
+    local mode="$1"  # Mode should be 'download' or 'view' #
+    local newUpdateVerStr=""
+    local wgetLogFile  changeLogFile  changeLogTag  changeLogURL
 
     # Create directory to download changelog if missing
     if ! _CreateDirectory_ "$FW_BIN_DIR" ; then return 1 ; fi
 
-    changeLogTag="$(echo "$(nvram get buildno)" | grep -qE "^386[.]" && echo "386" || echo "NG")"
-    if [ "$changeLogTag" = "386" ]
+    if [ "$mode" = "view" ]
```

Comment:
I need to come back to this after a coffee or something, I feel I am missing something as to why the duplicate code for view and download.

Reply:
Ah, I see below the reason now, you fee the version number into the function with a "download" mode.